substrate/frame/scheduler/src/lib.rs

```diff
@@ -440,6 +462,86 @@ pub mod pallet {
 			)?;
 			Ok(())
 		}
+
+		/// Set a retry configuration for a task so that, in case its scheduled run fails, it will
+		/// be retried after `period` blocks, for a total amount of `retries` retries or until it
+		/// succeeds.
+		///
+		/// If a retried task runs successfully before running out of retries, its remaining retry
+		/// counter will be reset to the initial value. If a retried task runs out of retries, it
+		/// will be removed from the schedule.
+		///
+		/// Tasks which need to be scheduled for a retry are still subject to weight metering and
+		/// agenda space, same as a regular task. Periodic tasks will have their periodic schedule
+		/// put on hold while the task is retrying.
+		#[pallet::call_index(6)]
+		#[pallet::weight(<T as Config>::WeightInfo::set_retry(T::MaxScheduledPerBlock::get()))]
+		pub fn set_retry(
+			origin: OriginFor<T>,
+			when: BlockNumberFor<T>,
+			index: u32,
+			retries: u8,
+			period: BlockNumberFor<T>,
+		) -> DispatchResult {
+			T::ScheduleOrigin::ensure_origin(origin.clone())?;
+			let origin = <T as Config>::RuntimeOrigin::from(origin);
+			let agenda = Agenda::<T>::get(when);
+			match agenda.get(index as usize) {
+				Some(Some(task)) =>
+					if matches!(
+						T::OriginPrivilegeCmp::cmp_privilege(origin.caller(), &task.origin),
+						Some(Ordering::Less) | None
+					) {
+						return Err(BadOrigin.into())
+					},
+				_ => return Err(Error::<T>::NotFound.into()),
+			}
+			Retries::<T>::insert(
+				(when, index),
```

Comment:
What happens when a retry is already scheduled and the retry config is changed?  
Do you think its important to be able to change the retry config?

Reply:
> What happens when a retry is already scheduled and the retry config is changed?

The next time the task comes up to be serviced the retry config would be updated. The current implementation allows you to change the config on the fly (e.g. you want to increase the period because something else in the system changed, but you don't want to deregister then register the task again).

> Do you think its important to be able to change the retry config?

I think it's useful but not necessary. That being said, I don't see a reason to forbid it. You pay for each execution either way, and it is simpler than to keep a separate state of the retry config per attempt just to restrict this behavior.

Reply:
Okay. Its not an issue, just wanted to hear your reasoning for it 😄